plugin/src/main/groovy/com/novoda/gradle/command/AdbTask.groovy

```diff
@@ -57,8 +57,14 @@ public class AdbTask extends org.gradle.api.DefaultTask {
         def adb = getAdb() ?: resolveFromExtension('adb')
         def deviceId = getDeviceId() ?: resolveFromExtension('deviceId')
         AdbCommand command = [adb: adb, deviceId: deviceId, parameters: parameters]
+
         logger.info "running command: $command"
-        handleCommandOutput(command.execute().text)
+
+        Process process = command.execute()
+        if (process.waitFor() != 0) {
```

Comment:
Exactly. But it doesn't change any backward compatibility because it was always `0` before.